### PR TITLE
feat(in-page-banner): rename content-group-banner to in-page-banner

### LINKED ISSUE #10754: [Component] Content group banner
Component review notes

### Change log
- [x] Rename component to `in page banner`
- [x] Add background color option

### Specs
- Figma spec file TBD

### PR DESCRIPTION
**Pull request by annawen1**
### Related Ticket(s)

Closes #10754

### Description

Renaming `content-group-banner` to `in-page-banner` + add background color options: "layer", "purple", "cyan".

This PR also renames the color tokens that were initially used in `callout` to more generic names as they are being used across a few components.

### Changelog

**New**

- add new background options: "layer", "cyan", "purple"

**Changed**

- renamed `content-group-banner` to `in-page-banner`

### CODE REVIEW

File: packages/web-components/examples/codesandbox/components-react/content-group-banner/src/index.js
Comment (ariellalgilmore) [inline packages/web-components/examples/codesandbox/components-react/content-group-banner/src/index.js:1]: can we change the folder name of this too `in-page-banner`

File: packages/web-components/src/components/in-page-banner/in-page-banner.scss [hunk 1]
```diff
@@ -7,8 +7,9 @@
...
 @use '@carbon/ibmdotcom-styles/scss/globals/vars' as *;
 @use '@carbon/ibmdotcom-styles/scss/internal/content-block';
-@use '@carbon/ibmdotcom-styles/scss/components/content-group-banner';
+// @use '@carbon/ibmdotcom-styles/scss/components/in-page-banner';
+@use '../../../../styles/scss/components/in-page-banner';
 
-:host(#{$c4d-prefix}-content-group-banner) {
+:host(#{$c4d-prefix}-in-page-banner) {
   display: block;
 }
```
Comment (ariellalgilmore) [inline packages/web-components/src/components/in-page-banner/in-page-banner.scss:15]: if we move the styles to the styles package, then we can just have 
`@use '@carbon/ibmdotcom-styles/scss/components/in-page-banner';`

**Review by ariellalgilmore — COMMENTED** [review]
looking good! just a few small comments:

When changing the cta's the `visually-hidden` section appears cause it's still using `bx` not `cds`
Screenshot 2023-10-03 at 12 23 44 PM

File: packages/web-components/src/components/in-page-banner/in-page-banner.scss (hunk 1 above)
Comment (annawen1) [reply packages/web-components/src/components/in-page-banner/in-page-banner.scss:15]: oh whoops yep

**Review by RichKummer — COMMENTED** [review]
Hey @annawen1 , the new background look great! For the `purple` and `cyan` options, can you bring in the `cds-callout-color-purple` and `cds-callout-color-cyan` and apply those to the appropriate headings?

This way those colors will be consistent with what we have in the `callout` components.

Screenshot 2023-10-04 at 2 57 56 PM

**Review by RichKummer — APPROVED** [review]
Looks great @annawen1 , thank you for making the token updates! 💪

**Review by ariellalgilmore — APPROVED** [review]
Everything LGTM!!! just need to update the examples to not use the `link-list-item-cta`

**Comment by annawen1** [comment]
The storybook and storybook docs examples have been updated, I'll update the codesandbox example in a new PR!

**Pull request merged**